src/datasets/config.py

```diff
@@ -202,6 +202,7 @@
 LICENSE_FILENAME = "LICENSE"
 METRIC_INFO_FILENAME = "metric_info.json"
 DATASETDICT_JSON_FILENAME = "dataset_dict.json"
+METADATA_CONFIGS_FIELD = "builder_config"
```

Comment:
The first one is OK (if we use default: true for consistency across elements of the list)

The second one might raise issues: we don't support yaml field that may contain dots - and config names can contain dots

Reply:
right now we allow having one config and have the YAML be a single struct instead of a list of one struct, but I feel like it's a bit weird when "configs" is in plural. So I would either keep it singular or always require a list ideally. Not sure how hard it would be to change that though
```
configs:
   data_files: data/*
```

Reply:
`config` vs. `configs` - IMO we should choose the final value based on the most popular (anticipated) use case: using the YAML to define multiple configs or to change the default config params' values (the single config case)

Reply:
I think the most anticipated use case will be push_to_hub with one single config

Reply:
What's the point in using `push_to_hub` specifying a config but with only one config?

Reply:
I meant when someone used push_to_hub without specifying a config

Reply:
I mean it was developed to allow people to specify configurations in cases when it wasn't possible before. For this to be possible and not braking anything, default `push_to_hub` had to be adjusted a bit to be aligned but I think it doesn't mean that we should take this use case as a default one for the feature that is supposed to be used differently. Does it make sense? 
By the way, in this context @mariosasko 's idea of having other type of file for `push_to_hub` looks valid to me, since the file/part of file with configuration is created automatically and not supposed to be modified manually. We can preserve the same format (yaml - to process it in the same way in the code) but make it less "notable" for users by prepending the filename with a dot lol. This way it (arguably) won't look weird weather it has `config` or `configs` inside because it's like not supposed to be touched at all.